### ISSUE TITLE
'*** -[__NSSetM addObject:]: object cannot be nil' error caused by a decoration view

### ISSUE THREAD

**Issue by edelibasi** [TextureGroup/Texture#495]
*I have the flow layout that contains the decoration view in the first section:*

`public class SeparatorCollectionViewFlowLayout: UICollectionViewFlowLayout {

    init(separatorWidth: CGFloat = 1, separatorColor: UIColor = UIColor.black) {
        self.separatorWidth = separatorWidth
        self.separatorColor = separatorColor
        super.init()
        register(SeparatorView.self, forDecorationViewOfKind: "TimelineSeparator")

    }
    
    required public init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
    }
    
    override public func layoutAttributesForDecorationView(ofKind elementKind: String, at indexPath: IndexPath) -> UICollectionViewLayoutAttributes? {
        var frame = CGRect.zero

        if let rows = collectionView?.numberOfItems(inSection: 0), indexPath.row < rows {
            if let cellAttributes = layoutAttributesForItem(at: indexPath) {
                frame = cellAttributes.frame
            }
        }

        let layoutAttributes = ColoredViewLayoutAttributes(forDecorationViewOfKind: elementKind, with: indexPath)
        layoutAttributes.frame = CGRect(x: frame.minX, y: frame.maxY, width: frame.width, height: separatorWidth)
        layoutAttributes.zIndex = -1
        if let rows = collectionView?.numberOfItems(inSection: 0), indexPath.row >= rows {
            layoutAttributes.color = UIColor.clear
        } else {
            layoutAttributes.color = separatorColor
        }
        print(layoutAttributes)
        print(layoutAttributes.description)
        print(layoutAttributes.debugDescription)
        
        return layoutAttributes
    }
    
    override public func layoutAttributesForElements(in rect: CGRect)
        -> [UICollectionViewLayoutAttributes]? {
            guard let baseLayoutAttributes = super.layoutAttributesForElements(in: rect) else {
                return nil
            }
            
            var layoutAttributes = baseLayoutAttributes
            baseLayoutAttributes.filter { $0.representedElementCategory == .cell }.forEach {(layoutAttribute) in
                
                if let attr = layoutAttributesForDecorationView(ofKind: "TimelineSeparator", at: layoutAttribute.indexPath) {
                    layoutAttributes.append(attr)
                }
            }
            
            return layoutAttributes
    }
}

private class ColoredViewLayoutAttributes: UICollectionViewLayoutAttributes {
    var color: UIColor = UIColor.clear
}

private class SeparatorView: _ASCollectionReusableView {

    override func apply(_ layoutAttributes: UICollectionViewLayoutAttributes) {
        super.apply(layoutAttributes)

        guard let coloredLayoutAttributes = layoutAttributes as? ColoredViewLayoutAttributes else {
            return
        }
        backgroundColor = coloredLayoutAttributes.color
    }
}

*and the location where it crashes: (ASCollectionViewLayoutController.m)*

```
- (NSSet<ASCollectionElement *> *)elementsWithinRangeBounds:(CGRect)rangeBounds map:(ASElementMap *)map
{
  NSArray *layoutAttributes = [_collectionViewLayout layoutAttributesForElementsInRect:rangeBounds];
  NSMutableSet<ASCollectionElement *> *elementSet = [NSMutableSet setWithCapacity:layoutAttributes.count];
  
  for (UICollectionViewLayoutAttributes *la in layoutAttributes) {
    // Manually filter out elements that don't intersect the range bounds.
    // If a layout returns elements outside the requested rect this can be a huge problem.
    // For instance in a paging flow, you may only want to preload 3 pages (one center, one on each side)
    // but if flow layout includes the 4th page (which it does! as of iOS 9&10), you will preload a 4th
    // page as well.
    if (CATransform3DIsIdentity(la.transform3D) && CGRectIntersectsRect(la.frame, rangeBounds) == NO) {
      continue;
    }
    [elementSet addObject:[map elementForLayoutAttributes:la]];
  }

  return elementSet;
}
```
*I noticed the crash happens when the app is launched after it was in background for a while*

**Comment by Adlai-Holler** [comment]
I believe this was fixed in version 2.4, when we migrated from NSMutableSet in this method to NSHashTable, which accepts nil. Could you try the latest version and see if the issue still exists?

**Comment by Adlai-Holler** [comment]
@edelibasi Feel free to reopen this if I'm mistaken, but @mlilek-nyt just confirmed this is resolved in latest. Cheers man!
